wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/parser/ProtoParserTest.kt

```diff
@@ -2429,16 +2422,62 @@ class ProtoParserTest {
                 location = location.at(1, 1),
                 name = "Foo",
                 constants = listOf(
-                    EnumConstantElement(location.at(2, 3), "enum", 0),
-                    EnumConstantElement(location.at(3, 3), "service", 1),
-                    EnumConstantElement(location.at(4, 3), "message", 2),
+                    EnumConstantElement(location.at(2, 3), "syntax", 0),
+                    EnumConstantElement(location.at(3, 3), "import", 1),
+                    EnumConstantElement(location.at(4, 3), "package", 2),
+                    EnumConstantElement(location.at(7, 3), "message", 5,
+                        documentation = "option = 3;\nreserved = 4;"),
+                    EnumConstantElement(location.at(8, 3), "enum", 6),
+                    EnumConstantElement(location.at(9, 3), "service", 7),
+                    EnumConstantElement(location.at(10, 3), "extend", 8),
+                    EnumConstantElement(location.at(11, 3), "rpc", 9),
+                    EnumConstantElement(location.at(12, 3), "oneof", 10),
+                    EnumConstantElement(location.at(13, 3), "extensions", 11),
+                )
+            )
+        )
+    )
+    assertThat(ProtoParser.parse(location, proto)).isEqualTo(expected)
+  }
+
+  @Test fun protoKeywordAsMessageNameAndField() {
+    // Note: cases were protoc fails are commented out.
+    val proto = """
+      |message message {
+      |  optional message message = 1;
```

Comment:
what about proto3?

Reply:
this is the one below commented out, it fails on both protoc/wire